tools/queries.cpp

```
@@ -44,85 +44,156 @@
 using namespace pisa;
 using ranges::views::enumerate;
 
-template <typename Fn>
-void extract_times(
-    Fn fn,
-    std::vector<Query> const& queries,
-    std::vector<Score> const& thresholds,
-    std::string const& index_type,
-    std::string const& query_type,
-    size_t runs,
-    std::ostream& os
+enum class AggregationType { None = 0, Min = 1, Mean = 2, Median = 3, Max = 4 };
+
+[[nodiscard]] auto to_string(AggregationType type) -> std::string {
+    switch (type) {
+    case AggregationType::None: return "none";
+    case AggregationType::Min: return "min";
+    case AggregationType::Mean: return "mean";
+    case AggregationType::Median: return "median";
+    case AggregationType::Max: return "max";
+    }
+    return "unknown";
```

This should not return a value. Let's throw an exception instead. Maybe `std::logic_error`?